lib/utils.js

```diff
@@ -150,3 +151,19 @@ function headsToVersion (heads) {
 
   return Buffer.concat(bufAccum)
 }
+
+// From: https://stackoverflow.com/questions/4459928/how-to-deep-clone-in-javascript/
```

Comment:
I played with this code and [it has bugs](https://stackoverflow.com/questions/4459928/how-to-deep-clone-in-javascript/#comment117527575_34624648). Could you use the [`clone`](https://github.com/pvorb/clone) module instead? In general, I tend to look on npm or [node-modules](http://node-modules.com/) for well-tested solutions before copying code off StackOverflow.

Reply:
Wasn't aware of node-modules, thanks :) Will change to use `clone`.